### ISSUE TITLE
.findOne() requires a criteria error when updating models using couchdb

### ISSUE THREAD

**Issue by tmjam** [balderdashy/sails#3029]
I am using couchdb adapter with sails js. Updating model's with primary key set as string gives the following error.

.findOne() requires a criteria

The problem I saw was the Model.update function does not return the updated object with its primary key and hence fails the lookup on the following line.
https://github.com/balderdashy/sails/blob/master/lib/hooks/blueprints/actions/update.js#L87

The quickest solution i found was to update the above line with 
<code> var Q = Model.findOne(pk);</code>

**Comment by tjwebb** [comment]
I think the couchdb adapter repo might be a better place to file this.

**Comment by iriscyo** [comment]
Hello @tmjam , have you defined the primary key in your model?

**Comment by sailsbot** [comment]
Thanks for posting, @tmjam.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by markmssd** [comment]
Heads up!

I'm also running into the same problem. I am using a custom adapter for Parse.com

When trying to update a model, it throws that error. Node that the model gets successfully updated into Parse.

My ```id``` attribute is defined as:
```
// custom id to conform to Parse's objectId field
id: {
  columnName: 'objectId',
  type: 'string',
  primaryKey: true,
  unique: true
}
```

Noteworthy: I am also setting ```autoPK``` to ```false```

@tmjam @tjwebb any advice would be greatly appreciated!
